src/Passes/docs/src/DeveloperGuide/CodeQuality.md

```diff
@@ -0,0 +1,167 @@
+unit test passes and that there are no erros found by the static analyser.
+
+To setup the CI environment, run following commands
+
+```sh
+source develop.env
+virtualenv develop__venv
+source develop__venv/bin/activate
+pip install -r requirements.txt
+```
+
+These adds the necessary environment variables to ensure that you have the `TasksCI` package and all required dependencies.
+
+To check the style, run
+
+```sh
+./manage stylecheck
+```
+
+To test that the code compiles and tests passes run
+
+```sh
+./manage test
+```
+
+Finally, to analyse the code, run
+
+```sh
+./manage lint
+```
+
+You can run all processes by running:
+
+```sh
+./manage runci
+```
+
+As `clang-tidy` and `clang-format` acts slightly different from version to version and on different platforms, it is recommended
+that you use a docker image to perform these steps. TODO(TFR): The docker image is not added yet and this will be documented in the future.
+
+## Running tests
+
+In order to run the tests, you first need to build the library. Assuming that this is already done and the corresponding build is in `Debug/`, run the tests from the `Debug` folder:
+
+```
+lit tests/ -v
+-- Testing: 2 tests, 2 workers --
+PASS: Quantum-Passes :: QirAllocationAnalysis/case1.ll (1 of 2)
+PASS: Quantum-Passes :: QirAllocationAnalysis/case2.ll (2 of 2)
+
+Testing Time: 0.27s
+  Passed: 2
+```
+
+The C++ test suite can also be ran from the debug by first building all targets:
+
+```sh
+cmake ..
+make
+```
+
+and then running following command:
+
+```sh
+ctest
+```
+
+## Writing C++ tests
+
+TODO(tfr): Write this section
+
+## Modifying code quality requirements
+
+This component is the largest part of this PR. The continuous integration component includes:
+
+1. Style formatting to ensure that everything looks the same. This includes checking that relevant copyrights are in place.
+2. Static analysis
+3. Unit testing
+
+The automatic style enforcement is configurable with the ability to easily add or remove rules. Currently the source pipelines are defined as:
+
+```python
+SOURCE_PIPELINES = [
+    {
+        "name": "C++ Main",
+        "src": path.join(PROJECT_ROOT, "Source"),
+
+        "pipelines": {
+            "hpp": [
+                require_pragma_once,
+                enforce_cpp_license,
+                enforce_formatting
+            ],
+            "cpp": [
+                enforce_cpp_license,
+                enforce_formatting
+            ]
+        }
+    },
+    # ...
+]
+```
+
+This part defines pipelines for `.hpp` files and `.cpp` files allowing the developer to add such requirements such as having copyright in the op of the source file and ensure that formatting follows that given by `.clang-format`.
+
+Each of these CI stages can executed individually using `./manage` or you can run the entire CI process by invoking `./manage runci`. An example of what this may look like is here:
```

Comment:
```suggestion
Each of these CI stages can be executed individually using `./manage` or you can run the entire CI process by invoking `./manage runci`. An example of what this may look like is here:
```